src/main/java/ca/team2706/frc/robot/commands/drivebase/DriverAssistVision.java

```diff
@@ -0,0 +1,456 @@
+            if ((angRobotHeadingCurrent_Field >= 0.0 && angRobotHeadingCurrent_Field <= 45.0) || (angRobotHeadingCurrent_Field >= 315.0 && angRobotHeadingCurrent_Field < 360.0)) {
+                angRobotHeadingFinal_Field = 0.0;
+            } else if (angRobotHeadingCurrent_Field >= 45.0 && angRobotHeadingCurrent_Field < 135.0) {
+                angRobotHeadingFinal_Field = 90.0;
+            } else if (angRobotHeadingCurrent_Field >= 135.0 && angRobotHeadingCurrent_Field < 225.0) {
+                angRobotHeadingFinal_Field = 180.0;
+            } else if (angRobotHeadingCurrent_Field >= 225.0 && angRobotHeadingCurrent_Field < 315.0) {
+                angRobotHeadingFinal_Field = 270.0;
+            }
+        }
+        else if (driverAssistRocket == true) {
+            // Assist request for rocket ship targets
+            Log.d("DAV: Assist for rocket requested");
+
+            // Rockets on left
+            if (angRobotHeadingCurrent_Field >= 90.0 && angRobotHeadingCurrent_Field <= 150.0) {
+                angRobotHeadingFinal_Field = 120.0;
+            } else if (angRobotHeadingCurrent_Field >= 150.0 && angRobotHeadingCurrent_Field <= 210.0) {
+                angRobotHeadingFinal_Field = 180.0;
+            } else if (angRobotHeadingCurrent_Field >= 210.0 && angRobotHeadingCurrent_Field <= 270.0) {
+                angRobotHeadingFinal_Field = 240.0;
+            } 
+            // Rockets on right
+            else if (angRobotHeadingCurrent_Field >= 30.0 && angRobotHeadingCurrent_Field <= 90.0) {
+                angRobotHeadingFinal_Field = 60.0;
+            } else if ((angRobotHeadingCurrent_Field >= 0.0 && angRobotHeadingCurrent_Field <= 30.0) || (angRobotHeadingCurrent_Field >= 330.0 && angRobotHeadingCurrent_Field <= 360.0)) {
+                angRobotHeadingFinal_Field = 0.0;
+            } else if (angRobotHeadingCurrent_Field >= 270.0 && angRobotHeadingCurrent_Field <= 330.0) {
+                angRobotHeadingFinal_Field = 300.0;
+            }
+        }
+        Log.d("DAV: angRobotHeadingFinal_Field: " + angRobotHeadingFinal_Field);
+
+        return angRobotHeadingFinal_Field;
+    }
+
+    /**
+     * Returns previously computed final desired angle of robot heading respect to field frame.
+     * (For testing purposes.)
+     * 
+     * @return Previously computed final desired angle of robot heading respect to field frame in degrees
+     */
+    public double getAngRobotHeadingFinal_Field() {
+        return angRobotHeadingFinal_Field;
+    }
+
+    /** 
+     * Angle of robot heading with respect to x axis of field frame
+     * 
+    */
+    private double angRobotHeadingFinal_Field = 0.0;
+
+    /**  
+     * Value of angYawTargetWrtCameraLOSCWposAngle (yaw angle to target wrt camera line of sight in degrees)
+     * from the previous time the command was issued.
+     * 
+    */
+    private double angYawTargetWrtCameraLOSCWposPrev = 0.0;
+
+    private boolean commandAborted;
```

Comment:
Generally preferred to have fields defined near the top.

Reply:
All private fields moved to the top